spring-integration-ip/src/main/java/org/springframework/integration/ip/tcp/connection/TcpConnectionSupport.java

```diff
@@ -253,13 +260,33 @@ public void registerListener(TcpListener listener) {
 		// Determine the actual listener for this connection
 		if (!(this.listener instanceof TcpConnectionInterceptor)) {
 			this.actualListener = this.listener;
-		} else {
+		}
+		else {
 			TcpConnectionInterceptor outerInterceptor = (TcpConnectionInterceptor) this.listener;
 			while (outerInterceptor.getListener() instanceof TcpConnectionInterceptor) {
 				outerInterceptor = (TcpConnectionInterceptor) outerInterceptor.getListener();
 			}
 			this.actualListener = outerInterceptor.getListener();
 		}
+		this.listenerRegisteredLatch.countDown();
+	}
+
+	/**
+	 * Set whether or not automatic or manual registratiom of the {@link TcpListener} is to be
+	 * used. (Default automatic). When manual registration is in place, incoming messages will
+	 * be delayed until the listener is registered.
+	 * @since 1.4.5
+	 */
+	public void enableManualListenerRegistration() {
+		this.manualListenerRegistration = true;
+		this.listener = new TcpListener() {
```

Comment:
Yes, I thought about that after realizing it was used at runtime via `getListener()`, but I decided to leave it in case someone ever accesses the field directly.